docs/src/userguide/extension_types.rst

```diff
@@ -416,22 +630,19 @@ method using the usual Python technique, i.e.::
 
     Parrot.describe(self)
 
-`cdef` methods can be declared static by using the @staticmethod decorator.
+:keyword:`cdef`/``@ccall`` methods can be declared static by using the ``@staticmethod`` decorator.
 This can be especially useful for constructing classes that take non-Python
-compatible types.::
+compatible types.:
+
+.. tabs::
 
-    cdef class OwnedPointer:
-        cdef void* ptr
+    .. group-tab:: Pure Python
 
-        def __dealloc__(self):
-            if self.ptr is not NULL:
-                free(self.ptr)
+        .. literalinclude:: ../../examples/userguide/extension_types/owned_pointer.py
 
-        @staticmethod
-        cdef create(void* ptr):
-            p = OwnedPointer()
-            p.ptr = ptr
-            return p
```

Comment:
This would normally be a classical example for a `@classmethod` and not a `@staticmethod`. I hope there is a test that this actually works (not just compiles but actually runs correctly), but since documentation should always aim to show best practices, I would suggest switching to `@classmethod` here and using the `cls` argument for the instantiation to nicely support subclassing.

Accordingly, the longer example below about factory methods would be a good candidate for `@classmethod` as well, but since it also shows the usage of `.__new__()`, which is only fast when the types are known at compile time, I'm not sure if it should. Not sure what Cython does for `cls.__new__(cls)` – probably nothing special, but worth testing out. Could be a missed opportunity for an optimisation in class methods.

Reply:
It seems that `@classmethod` is not supported with `cdef` methods:
```
cython -3a owned_pointer.pyx

Error compiling Cython file:
------------------------------------------------------------
...
    def __dealloc__(self):
        if self.ptr is not NULL:
            free(self.ptr)


    @classmethod
   ^
------------------------------------------------------------

owned_pointer.pyx:13:4: Cdef functions cannot take arbitrary decorators.

Error compiling Cython file:
------------------------------------------------------------
...


    @classmethod
    cdef create(cls, void* ptr):
        p = cls()
        p.ptr = ptr
               ^
------------------------------------------------------------

owned_pointer.pyx:16:16: Cannot convert 'void *' to Python object
```

Reply:
Ok, good to know. It should be possible to implement C class methods but would require a bit of work on the caller side since I think it should be the callers responsibility to pass the right class. And that means that the caller would have to know that it's calling a class method.

Reply:
I have added note about that to documentation.